torch/tensor.py

```diff
@@ -41,6 +41,13 @@ def __reduce_ex__(self, proto):
         check_serializing_named_tensor(self)
         # See Note [Don't serialize hooks]
         torch.utils.hooks.warn_if_has_hooks(self)
+        if self.device.type == 'xla':
+            args = (self.cpu().tolist(),
```

Comment:
Are you *sure* you want a `tolist()` call here? It sounds very slow to me :)

Reply:
@ezyang There's actually one tricky implementation detail in pytorch serialization that prevents this to be a CPU tensor. 
If we return `self.cpu()` here, under the hood pytorch will serialize the storage of CPU tensor. But when we reconstruct in `torch.load`, we reconstruct the storage with randomly initialized data and update the storage afterward(which also updated the tensor if it were CPU/CUDA). 
But in XLA case, `.to` in `_rebuild_xla_tensor` is called before the storage update so XLA tensor doesn't get the updated data.

Reply:
If we want `reduce_ex` to use CPU as intermediate, we might need to hack in `torch.load` to mandate the storage update as well because this is no-op for XLA tensor. https://github.com/pytorch/pytorch/blob/master/torch/serialization.py#L620
Since we'll revisit the  serialization solution afterwards, I just want to keep this change simple and not hack anymore :P

Reply:
OK, I understand. The hack seems justified, but this deserves a comment in the source code.